### PR TITLE
Add recipe

### PR DESCRIPTION
**Pull request by Marie-Sail**

### CODE REVIEW

File: backend/src/app.js [hunk 1]
```diff
@@ -6,7 +6,9 @@ const express = require("express");
 
 const app = express();
 const cors = require("cors");
+/* const path = require("path"); */
```
Comment (Thomas-Nigon) [inline backend/src/app.js:9]: dead code

File: backend/src/controllers/stepControllers.js [hunk 2]
```diff
@@ -36,7 +36,25 @@ const browse = async (req, res, next) => {
   }
 };
 
+const add = async (req, res, next) => {
+  // Extract the item data from the request body
+  const step = req.body;
+  console.warn("step:", step);
```
Comment (Thomas-Nigon) [inline backend/src/controllers/stepControllers.js:42]: remove warn

File: backend/src/router.js [hunk 3]
```diff
@@ -35,7 +38,7 @@ router.get("/tag", tagControllers.browseByTag);
 router.get("/user", userControllers.browse);
 router.post("/adduser", hashPassword, userControllers.register);
 router.post("/login", authControllers.login);
-router.get("/logout", authControllers.logout);
+router.get("/logouts", authControllers.logout);
```
Comment (Thomas-Nigon) [inline backend/src/router.js:41]: remove s to logout

File: frontend/src/components/AddIngredients/AddIngredients.jsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import { useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+import PropTypes, { shape } from "prop-types";
+import "./AddIngredients.scss";
+
+function AddIgredients({ ingredientArr, setIngredientArr }) {
+  const [ingredientsBdd, setIngredientsBdd] = useState();
+  //   const [ingredientArr, setIngredientArr] = useState([]);
```
Comment (Thomas-Nigon) [inline frontend/src/components/AddIngredients/AddIngredients.jsx]: remove comment

File: frontend/src/pages/Search/Search.jsx [hunk 5]
```diff
@@ -1,18 +1,39 @@
...
   const [tags, setTags] = useState();
+
   useEffect(() => {
     fetch(`${import.meta.env.VITE_BACKEND_URL}/api/tag`)
       .then((response) => response.json())
       .then((data) => setTags(data))
       .catch((error) => console.error(error));
   }, []);
 
+  // Example POST method implementation:
```
Comment (Thomas-Nigon) [inline frontend/src/pages/Search/Search.jsx]: delete dead code

File: backend/src/router.js [hunk 6]
```diff
@@ -35,7 +38,7 @@ router.get("/tag", tagControllers.browseByTag);
 router.get("/user", userControllers.browse);
 router.post("/adduser", hashPassword, userControllers.register);
 router.post("/login", authControllers.login);
-router.get("/logout", authControllers.logout);
```
Comment (NemoOnTheNet) [inline backend/src/router.js:38]: changed required here

File: frontend/src/components/SelectCountry/SelectCountry.jsx [hunk 7]
```diff
@@ -1,11 +1,21 @@
 import "./SelectCountry.scss";
+import { useRef } from "react";
+import PropTypes from "prop-types";
 
-function SelectCountry() {
+function SelectCountry({ setCountry }) {
+  const countryRef = useRef();
+  const handleChange = () => {
+    setCountry(countryRef.current.value);
+    // console.log(countryRef.current.value);
```
Comment (NemoOnTheNet) [inline frontend/src/components/SelectCountry/SelectCountry.jsx:9]: dead code

File: frontend/src/components/SelectCountry/SelectCountry.jsx [hunk 8]
```diff
@@ -1,11 +1,20 @@
...
-      <option value="France" selected="selected">
-        France{" "}
-      </option>
+    <select
+      className="selectCountry"
+      name="country"
+      ref={countryRef}
+      onChange={handleChange}
+    >
+      <option defaultValue=""> Origine </option>
```
Comment (Thomas-Nigon) [inline frontend/src/components/SelectCountry/SelectCountry.jsx:18]: delete space?

File: frontend/src/pages/Search/Search.jsx [hunk 9]
```diff
@@ -32,8 +34,8 @@ function Search() {
         {tags
           ? tags.map((tag) => <TagCard key={tag.id} tag={tag} />)
           : "loading"}
-      </section>
-    </section>
+      </section>{" "}
```
Comment (Thomas-Nigon) [inline frontend/src/pages/Search/Search.jsx:37]: a virer?

File: backend/src/controllers/recipeControllers.js [hunk 10]
```diff
@@ -1,5 +1,30 @@
...
+const add = async (req, res, next) => {
+  // Extract the item data from the request body
+  const { title, time, dificulty, price, picture, diet, type, season } =
+    req.body.data;
+  try {
+    // Insert the item into the database
+    const insertId = await tables.recipe.create(
+      title,
+      time,
+      dificulty,
```
Comment (fantinerudent) [inline backend/src/controllers/recipeControllers.js:12]: difficulty*

**Review by NemoOnTheNet — APPROVED** [review]

**Review by Sax593 — APPROVED** [review]

**Pull request merged**